### ISSUE TITLE
#wait_readable and #wait_writable have different signature than the ruby methods

### ISSUE THREAD

**Issue by TiagoCardoso1983** [celluloid/celluloid-io#169]
Was inspecting http://ruby-doc.org/stdlib-2.0.0/libdoc/io/wait/rdoc/IO.html#method-i-wait_readable for removal of IO.select calls on some internal stuff. Both wait_* methods accept a timeout parameter. Also, #wait_readable is aliased to #wait . Don't know about the other ones.

**Comment by tarcieri** [comment]
Yeah, these are the new `io/wait` API introduced in Ruby 2.0, which Celluloid::IO predates.

I think we should try to support the upstream API.

**Comment by TiagoCardoso1983** [comment]
@tarcieri , either one wraps the #wait call in a Celluloid.timeout, or nio4r should provide support for timed-out IO monitors. Is option B possible?

**Comment by TiagoCardoso1983** [comment]
Shall we continue the discussion [here](https://github.com/celluloid/celluloid-io/pull/170)?
